hudi-common/src/main/java/org/apache/hudi/avro/HoodieAvroUtils.java

```diff
@@ -676,13 +676,15 @@ public static Schema getNestedFieldSchemaFromRecord(GenericRecord record, String
    */
   public static Schema getNestedFieldSchemaFromWriteSchema(Schema writeSchema, String fieldName) {
     String[] parts = fieldName.split("\\.");
-    int i = 0;
-    for (; i < parts.length; i++) {
+    Schema currentSchema = writeSchema;
```

Comment:
don't we have UTs for getNestedFieldSchemaFromWriteSchema ? 
can we add one and add some tests.

Reply:
SI functional test that I had added covers such a schema, but i've also add a separate UT for the util method that covers both positiv and negative scenarios.